src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_scans.rs

```diff
@@ -847,30 +848,47 @@ impl Default for ScanOptions {
     }
 }
 
+#[derive(Default)]
+pub struct ScanState {
+    pub accum: u64,
+    pub index: u64,
+    pub must_scan: bool,
+    pub checks: Vec<PossibleScanMatch>,
+}
+
+impl ScanState {
+    pub fn reset(&mut self) {
+        self.accum = 0;
+        self.index = 0;
+        self.must_scan = false;
+        self.checks.clear();
+    }
+
+    pub fn has_possible_matches(&self) -> bool { !self.possible_matches().is_empty() }
+
+    pub fn possible_matches(&self) -> &Vec<PossibleScanMatch> { &self.checks }
+}
+
 pub struct Scan {
     options: ScanOptions,
-    accum: u64,
-    index: u64,
+    stream_state: RefCell<ScanState>,
```

Comment:
>stream_state [](http://example.com/codeflow?start=4&length=12)

it's not a big deal but maybe `scan_states` is a better name here? because this object contains non-stream related data (like the accumulated checks)

Reply:
Maybe, that ScanState is only used for the "stream" interfaces.

I'm not sure what a good name is for them, I'm happy with scan_state, or internal_state.
It's not used at all when concurrent_* APIs are used by external callers.

Reply:
sure, how about `internal_state`, that seems nicely descriptive of what we've got here.